### DISCUSSION TITLE
How to resolve this issue while migrating the custom auth plugin built using emqx 4.0.x to 5.6.0

### DISCUSSION THREAD

**Discussion by venkatac-ayla** [emqx/emqx#12926] [Q&A]
We have a auth plugin which is working on 4.0.7 version of emqx. Now we need to make it compatible to work with emqx 5.6.0. While running the make we receive the following error. Could any one suggest how to overcome this issue?

**===> Error building application emqx:
     No project builder is configured for type mix**

**Compilation issue**
`===> Compiling c_src/async_queue.c
===> Compiling c_src/bcrypt.c
===> Compiling c_src/bcrypt_nif.c
===> Compiling c_src/blowfish.c
===> Linking /tmp/DNC-emqx-plugin/_build/default/lib/bcrypt/priv/bcrypt_nif.so
===> Compiling emqx
===> Error building application emqx:
     No project builder is configured for type mix
make: *** [Makefile:10: compile] Error 1
root@331d3ef24092:/tmp/DNC-emqx-plugin# cat rebar.config`

**Here is the rebar.config for the custom auth plugin**
`{deps, [
  {emqx, {git, "https://github.com/emqx/emqx.git", {tag, "v5.6.0"}}},
  qdate_localtime,
  {qdate, "0.5.0"},
  {jsone, "1.5.2"}
]}.

{erl_opts, [debug_info]}.`

**Comment by zmstone** [comment]
You can follow the docker images in this repo to setup your build environment:
https://github.com/emqx/emqx-builder

You can also pull the builder image to build the plugin:
```
$ ./show-latest-images.sh
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-alpine3.15.1
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-amzn2
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-amzn2023
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-debian10
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-debian11
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-debian12
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-debian9
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-el7
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-el8
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-el9
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-ubuntu16.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-ubuntu18.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-ubuntu20.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-25.3.2-2-ubuntu22.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-alpine3.15.1
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-amzn2
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-amzn2023
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-debian10
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-debian11
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-debian12
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-debian9
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-el7
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-el8
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-el9
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-ubuntu16.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-ubuntu18.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-ubuntu20.04
ghcr.io/emqx/emqx-builder/5.3-2:1.15.7-26.2.1-2-ubuntu22.04
```

**Comment by venkatac-ayla** [comment]
Thanks. Using erlang:25 docker image I am able to proceed
